### PR TITLE
blog: announce the new tilt extension api

### PR DESCRIPTION
**Pull request by nicks**
Hello @lianmakesthings, @nicksieger,

Please review the following commits I made in branch nicks/private:

aff865dc3e13837d717e1b8b81b24952aa851e3c (2022-02-24 16:56:53 -0500)
blog: announce the new tilt extension api

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by nicks** [comment]
this mostly cuts out some of the important bits from the general-purpose extension docs, and punches it up a bit in blog post form.

**Review by nicksieger — APPROVED** [review]
:shipit:

File: blog/_posts/2022-02-24-dev-env-libraries.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
+---
+slug: "dev-env-libraries"
+date: 2022-02-24
+author: nick
+layout: blog
+title: "Defining Your Own Library of Dev Env Patterns"
```
Comment (lianmakesthings) [inline blog/_posts/2022-02-24-dev-env-libraries.md:6]:
```suggestion
title: "Defining Your Own Pattern Library"
```
make it snazzy!

File: blog/_posts/2022-02-24-dev-env-libraries.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+```
+[url "ssh://git@github.com/"]
+	insteadOf = https://github.com/
+```
+
+## Sharing Extensions with the Community
+
+If you have an extension that you think would be generally useful, we love to
+see people contribute them back to the main repo!
```
Comment (lianmakesthings) [inline blog/_posts/2022-02-24-dev-env-libraries.md:127]:
```suggestion
see people contribute them back to the [main repo](https://github.com/tilt-dev/tilt-extensions)!
```

**Review by lianmakesthings — APPROVED** [review]
Awesome! 👏

**Pull request merged**